src/gwc/src/main/java/org/geoserver/gwc/config/GWCConfig.java

```diff
@@ -370,7 +362,7 @@ public boolean isEnabled(final String serviceId) {
             return isWMSCEnabled();
         }
         if ("wmts".equalsIgnoreCase(serviceId)) {
-            return isWMTSEnabled();
+            throw new ServiceException("To check if WMTS service is enable or disable use service info.");
```

Comment:
ServiceExceptions are to be used for OGC errors, this is an internal programming error instead no? You have moved the check for WMTS being enabled or not to a normal ServiceInfo, and this check should not be used anymore, right?